components/places/src/api/matcher.rs

```diff
@@ -259,15 +259,36 @@ struct OriginOrUrl<'query, 'conn> {
     conn: &'conn PlacesDb,
 }
 
+const WITH_ORIGIN_AUTOFILL_THRESHOLD: &str = r#"
+    WITH frecency_stats(count, sum, squares) AS (
+        SELECT
+            IFNULL(CAST((SELECT value FROM moz_meta WHERE key = "origin_frecency_count") AS REAL), 0.0),
+            IFNULL(CAST((SELECT value FROM moz_meta WHERE key = "origin_frecency_sum") AS REAL), 0.0),
+            IFNULL(CAST((SELECT value FROM moz_meta WHERE key = "origin_frecency_sum_of_squares") AS REAL), 0.0)
+    ),
+    autofill_frecency_threshold(value) AS (
+        SELECT
+            CASE count
```

Comment:
It might be slightly more readable to put this logic in a function, and change the fragment to something like `autofill_frecency_threshold((SELECT value FROM moz_meta WHERE key = "origin_frecency_count"), (SELECT value FROM moz_meta WHERE key = "origin_frecency_sum"), (SELECT value FROM moz_meta WHERE key = "origin_frecency_sum_of_squares"), :stddevMultiplier)`.

But the way you have it now matches Desktop, so it's your call.